appveyor.yml

```diff
@@ -178,20 +176,27 @@ test_script:
 
 # Flake8 Linting
  - ps: |
-    if($env:APPVEYOR_PULL_REQUEST_NUMBER) {
       $lintOutput = (Resolve-Path .\testOutput\lint\)
       $lintSource = (Resolve-Path .\tests\lint\)
+      $flake8Output = "$lintOutput\Flake8.txt"
       # When Appveyor runs for a pr,
       # the build is made from a new temporary commit,
       # resulting from the pr branch being merged into its base branch.
       # Therefore to create a diff for linting, we must fetch the head of the base branch.
       # In a PR, APPVEYOR_REPO_BRANCH points to the head of the base branch. 
-     git fetch -q origin $env:APPVEYOR_REPO_BRANCH
-     $flake8Output = "$lintOutput\PR-Flake8.txt"
+      # Additionally, we can not use a clone_depth of 1, but must use an unlimited clone.
+      if($env:APPVEYOR_PULL_REQUEST_NUMBER) {
+        git fetch -q --recurse-submodules origin $env:APPVEYOR_REPO_BRANCH

```

Comment:
Why do we need  to recursively fetch submodules here? Same question for the fetch further down. Linting does not require the submodules I would have thought...
I would say that fetching the submodules (now 3 times) would be most of the 40 seconds added to the build.

Reply:
I'm not convinced it does anything after further investigation. I thought it was due to possible issues with merging - but that should only be an issue on the try branch builds if anything, not in this condition as the checked out branch should already have master merged in. 

I think it might be easier to take it out and get a good example if any error is (still) thrown.

Reply:
The time difference is now much more negligible